contracts/contracts/buyback/BaseBuyback.sol

```diff
@@ -0,0 +1,266 @@
+
+    /**
+     * @dev Set address of Uniswap Universal Router for performing liquidation
+     * of platform fee tokens. Setting to 0x0 will pause swaps.
+     *
+     * @param _router Address of the Uniswap Universal router
+     */
+    function setUniswapUniversalRouter(address _router) external onlyGovernor {
+        _setUniswapUniversalRouter(_router);
+    }
+
+    function _setUniswapUniversalRouter(address _router) internal {
+        if (universalRouter != address(0)) {
+            // Remove previous router's allowance
+            IERC20(oToken).approve(universalRouter, 0);
+        }
+
+        universalRouter = _router;
+
+        emit UniswapUniversalRouterUpdated(_router);
+    }
+
+    /**
+     * @dev Sets the address that receives the OGV buyback rewards
+     * @param _address Address
+     */
+    function setRewardsSource(address _address) external onlyGovernor {
+        _setRewardsSource(_address);
+    }
+
+    function _setRewardsSource(address _address) internal {
+        require(_address != address(0), "Address not set");
+        rewardsSource = _address;
+        emit RewardsSourceUpdated(_address);
+    }
+
+    /**
+     * @dev Sets the address that can receive and manage the funds for Treasury
+     * @param _address Address
+     */
+    function setTreasuryManager(address _address) external onlyGovernor {
+        _setTreasuryManager(_address);
+    }
+
+    function _setTreasuryManager(address _address) internal {
+        require(_address != address(0), "Address not set");
+        treasuryManager = _address;
+        emit TreasuryManagerUpdated(_address);
+    }
+
+    /**
+     * @dev Swaps half of `oTokenAmount` to OGV
+     *      and the rest to CVX and finally lock up CVX
+     * @param oTokenAmount Amount of OUSD/OETH to swap
+     * @param minOGV Minimum OGV to receive for oTokenAmount/2
+     * @param minCVX Minimum CVX to receive for oTokenAmount/2
+     */
+    function swap(
+        uint256 oTokenAmount,
+        uint256 minOGV,
+        uint256 minCVX
+    ) external onlyGovernorOrStrategist nonReentrant {
+        _swap(oTokenAmount, minOGV, minCVX);
+    }
+
+    function _swap(
+        uint256 oTokenAmount,
+        uint256 minOGV,
+        uint256 minCVX
+    ) internal {
+        require(oTokenAmount > 0, "Invalid Swap Amount");
+        require(universalRouter != address(0), "Uniswap Router not set");
+        require(rewardsSource != address(0), "RewardsSource contract not set");
+        require(minOGV > 0, "Invalid minAmount for OGV");
+        require(minCVX > 0, "Invalid minAmount for CVX");
+
+        uint256 ogvBalanceBefore = IERC20(ogv).balanceOf(rewardsSource);
+        uint256 cvxBalanceBefore = IERC20(cvx).balanceOf(address(this));
+
+        // Build swap input
+        bytes[] memory inputs = new bytes[](2);
+
+        inputs[0] = abi.encode(
+            // Send swapped OGV directly to RewardsSource contract
+            rewardsSource,
+            oTokenAmount / 2,
```

Comment:
We use this `oTokenAmount / 2` calculation four times, computing it each time. Might be nice to put this into a variable(s) and then use that.

I don't think it matters practically here, but two `oTokenAmount / 2` may not equal same number as `oTokenAmount` because of rounding on each division. Usually if you are calculating some numbers that should make up a total in solidity, you calculate the first numbers, rounding down, and then send the all the rest remaining to the last:

```
// going for x + y == z
x = z / 5
y = z - x
```

Reply:
Turns out this does actually matter, in the super picky sense, because we send the full amount to router, but then don't use the rounding error to swap back to our coins. The amount is silly and tiny, but I like systems better when they are 100% correct.

Reply:
Yep, good catch, have made the change to fix this